src/main/java/smarthome/Room.java

```diff
@@ -0,0 +1,147 @@
+package smarthome;
+
+import java.util.Arrays;
+
+/**
+ * The type Room.
+ */
+public class Room {
+    /**
+     * The Id.
+     */
+    private int id;
+    /**
+     * The Name.
+     */
+    private String name;
+    /**
+     * The Windows.
+     */
+   private Window[] windows;
+    /**
+     * The Lights.
+     */
+   private Light[] lights;
+    /**
+     * The Doors.
+     */
+   private Door[] doors;
+
+    /**
+     * Instantiates a new Room.
+     *
+     * @param id      the id
+     * @param name    the name
+     * @param windows the windows
+     * @param lights  the lights
+     * @param doors   the doors
+     */
+    public Room(int id, String name, Window[] windows, Light[] lights, Door[] doors) {
```

Comment:
I won't ask you to change this, since it'll depend on how we implement the ``HouseLayout`` class, but a constructor with the signature ``public Room(int id, String name, int windows, int lights, int doors)``, and which creates its own Window, Light and Door arrays inside the constuctor would also make a lot of sense.